pkg/limits/ingest_limits.go

```diff
@@ -159,6 +186,19 @@ func (s *IngestLimits) TransferOut(_ context.Context) error {
 // The client is configured to consume stream metadata from a dedicated topic with the metadata suffix.
 func NewIngestLimits(cfg Config, logger log.Logger, reg prometheus.Registerer) (*IngestLimits, error) {
 	var err error
+
+	// If RateWindow is not set, default to 5 minutes to match common Prometheus rate() window
+	if cfg.RateWindow == 0 {
```

Comment:
This validation should be done in `cfg.Validate()` as we do elsewhere. We can expect configurations passed to `NewIngestLimits` to be validated and have usable values.

Reply:
Yes good case to move them up to the validation.